src/code/z_demo.c

```diff
@@ -443,15 +444,16 @@ void Cutscene_Command_Terminator(GlobalContext* globalCtx, CutsceneContext* csCt
 
     if ((gSaveContext.game_mode != 0) && (gSaveContext.game_mode != 3) && (globalCtx->sceneNum != SCENE_SPOT00) &&
         (csCtx->frames > 20) &&
-        (!~(globalCtx->state.input[0].padPressed | ~A_BUTTON) || !~(globalCtx->state.input[0].padPressed | ~B_BUTTON) ||
-         !~(globalCtx->state.input[0].padPressed | ~START_BUTTON)) &&
+        (!~(globalCtx->state.input[0].pressed_diff.input.button | ~A_BUTTON) ||
+         !~(globalCtx->state.input[0].pressed_diff.input.button | ~B_BUTTON) ||
+         !~(globalCtx->state.input[0].pressed_diff.input.button | ~START_BUTTON)) &&
```

Comment:
Is there any chance we could avoid parts of the long redundant union/struct accesses here? These are getting really long for a simple u16 access, and "input" is repeated multiple times which isn't really ideal imo.